common/build-style/python3-module.sh

```diff
@@ -24,6 +30,11 @@ do_build() {
 }
 
 do_check() {
+	if [ -n "$python_pep517" ]; then
+		msg_warn "No standard test exists for PEP517 Python templates"
```

Comment:
We could try adding the `pytest` option to `do_check`, what do you think?

`if command -v pytest >/dev/null; then \n pytest`

Reply:
Not a bad idea, but I've gone one step further. Running `python setup.py test` complains about deprectation and suggests `tox` instead. Because that is the official recommendation, I try `tox` first if possible, otherwise try `pytest` if possible, and finally fall back to the old behavior for non-PEP517 builds.

I've modified a few templates as an example of how the change eliminates some redundancy.

If there is not significant debate about the PEP517 changes, I'll squash the two `build-style/python3-module` commits before merging. For now, I'll keep them apart so we can separate the two issues if need be.

Reply:
See #25052 for some discussion about tox.

Reply:
Based on opposition in the mentioned issue, I dropped `tox` from the build style and kept the preference for `pytest` over the deprecated `setup.py test` method.